$DragonFly: src/gnu/usr.bin/groff/tmac/patches/troffrc.no_obj.patch,v 1.1 2005/09/01 20:47:14 asmodai Exp $

--- troffrc	2005-08-31 14:28:07.000000000 +0200
+++ troffrc	2005-08-31 15:56:52.000000000 +0200
@@ -19,6 +19,7 @@
 .do ds troffrc!latin1 tty.tmac
 .do ds troffrc!utf8 tty.tmac
 .do ds troffrc!cp1047 tty.tmac
+.do ds troffrc!koi8-r tty.tmac
 .do ds troffrc!lj4 lj4.tmac
 .do ds troffrc!lbp lbp.tmac
 .do ds troffrc!html html.tmac
@@ -29,7 +30,10 @@
 .
 .\" test whether we work under EBCDIC
 .do ie '\[char97]'a' \
-.	do tr \[char160]\~
+.      ie '\*[.T]'koi8-r' \
+.              do tr \[char154]\~
+.      el \
+.              do tr \[char160]\~
 .el \
 .	do tr \[char65]\~
 .
@@ -39,8 +43,18 @@
 .\" Load hyphenation patterns and exceptions.
 .do hpf hyphen.us
 .do hpfa hyphenex.us
+.if '\*[.T]'koi8-r' \
+.      do hpfa hyphen.ru
 .
 .\" Handle paper formats
 .do mso papersize.tmac
 .
+.\" Disable SGR support in grotty(1).
+.if n \{\
+.  do nop \X'tty: sgr 0'
+.  sp -1
+.  nr nl 0-1
+.  nr % -1
+.\}
+.
 .\" Don't let blank lines creep in here.
